accord-core/src/main/java/accord/utils/SmallBitSet.java

```diff
@@ -40,29 +40,50 @@ public long bits()
         return bits;
     }
 
+    private static long bitSafe(int i)
+    {
+        validateIndex(i);
+        return bit(i);
+    }
+
+    private static void validateIndex(int i)
+    {
+        if (i >= 64 || i < 0)
+            throw new IndexOutOfBoundsException("Unable to access bit " + i + "; must be between 0 and 63");
+    }
+
+    @Override
     public boolean set(int i)
     {
-        long bit = bit(i);
+        long bit = bitSafe(i);
         boolean result = 0 == (bits & bit);
         bits |= bit;
         return result;
     }
 
     @Override
-    public void setRange(int from, int to)
+    public void setRange(int fromInclusive, int toExclusive)
     {
-        bits |= ~(-1L << to) & (-1L << from);
+        validateIndex(fromInclusive);
+        Invariants.requireArgument(fromInclusive <= toExclusive, "from > to (%s > %s)", fromInclusive, toExclusive);
+        if (fromInclusive == toExclusive)
+            return;
+
+        bits |= (-1L >>> (64 - (toExclusive & 63))) & (-1L << (fromInclusive & 63));
     }
 
+    @Override
     public boolean get(int i)
     {
+        if (i >= 64) return false;
```

Comment:
shouldn't this be IIOBE?

Reply:
we spoke in slack, switched to IIOBE.  if `i == size()` then we accept and return `-1` for `nextSetBit` but `get` now rejects.

If `i > size()` we will reject for all input